src/Integration.UnitTests/MefServices/DeprecatedSonarRuleSetManagerTests.cs

```diff
@@ -0,0 +1,353 @@
+
+            var projectPath = Path.GetTempFileName();
+
+            var rulesetName = $"{Guid.NewGuid()}.ruleset";
+            var ruleset = TestRuleSetHelper.CreateTestRuleSetWithRuleIds(new[] { "S110" }, "SonarAnalyzer.Foo");
+            ruleset.WriteToFile(Path.Combine(Path.GetDirectoryName(projectPath), rulesetName));
+
+            var project = new ProjectMock(projectPath);
+            projectSystemHelper.Projects = new[] { project };
+            project.SetBuildProperty(Constants.CodeAnalysisRuleSetPropertyKey, rulesetName);
+
+            this.ruleSetProvider.RegisterProjectInfo(project, new RuleSetDeclaration(project, new PropertyMock("bar", null),
+                rulesetName, "", ""));
+            CreateTestSubject();
+
+            // Act
+            this.activeSolutionTracker.SimulateActiveSolutionChanged(isSolutionOpen: true);
+
+            // Assert
+            loggerMock.Verify(x => x.WriteLine(DeprecatedSonarRuleSetManager.DeprecationMessage), Times.Once);
+        }
+
+        [TestMethod]
+        public void OnSolutionBindingChanged_WhenSolutionInLegacyMode_DoesNotWarn()
+        {
+            // Arrange
+            this.activeSolutionBoundTracker.CurrentConfiguration = new NewConnectedMode.BindingConfiguration(
+                new Persistence.BoundSonarQubeProject(), NewConnectedMode.SonarLintMode.LegacyConnected);
+            CreateTestSubject();
+
+            // Act
+            this.activeSolutionBoundTracker.SimulateSolutionBindingChanged(new ActiveSolutionBindingEventArgs(
+                new NewConnectedMode.BindingConfiguration(new Persistence.BoundSonarQubeProject(),
+                NewConnectedMode.SonarLintMode.LegacyConnected)));
+
+            // Assert
+            loggerMock.Verify(x => x.WriteLine(DeprecatedSonarRuleSetManager.DeprecationMessage), Times.Never);
+        }
+
+        [TestMethod]
+        public void OnSolutionBindingChanged_WhenStandaloneAndProjectHasSonarRules_DoWarn()
+        {
+            // Arrange
+            this.activeSolutionBoundTracker.CurrentConfiguration = new NewConnectedMode.BindingConfiguration(
+                new Persistence.BoundSonarQubeProject(), NewConnectedMode.SonarLintMode.LegacyConnected);
+            var projectPath = Path.GetTempFileName();
+
+            var rulesetName = $"{Guid.NewGuid()}.ruleset";
+            var ruleset = TestRuleSetHelper.CreateTestRuleSetWithRuleIds(new[] { "S110" }, "SonarAnalyzer.Foo");
+            ruleset.WriteToFile(Path.Combine(Path.GetDirectoryName(projectPath), rulesetName));
+
+            var project = new ProjectMock(projectPath);
+            projectSystemHelper.Projects = new[] { project };
+            project.SetBuildProperty(Constants.CodeAnalysisRuleSetPropertyKey, rulesetName);
+
+            this.ruleSetProvider.RegisterProjectInfo(project, new RuleSetDeclaration(project, new PropertyMock("bar", null),
+                rulesetName, "", ""));
```

Comment:
This _create-and-register-ruleset_ code could be moved to a helper function (perhaps with a parameter to say whether the ruleset should contain a Sonar rule or not).